pennylane/compiler/compiler.py

```diff
@@ -41,17 +46,38 @@ class AvailableCompilers:
 
     # The dictionary of installed compiler packages
     # and their entry point loaders.
-    names_entrypoints = {}
+    names_entrypoints = defaultdict(dict)
+
+    # The map consists of supported compiler names (str) and their version compatibility (bool).
+    # This boolean indicates whether the installed version of a compiler package is greater
+    # than or equal to the minimum version.
+    # This value will be updated in `_check_compiler_version` to reduce the required
+    # version checks of installed compiler packages at runtime.
+    names_versions = defaultdict(dict)
```

Comment:
The caching is a nice feature 👍 

This is optional, but I would find a name for the dictionary such as `compiler_checked` more descriptive of what the code does (for instance a `false` Boolean does not indicate the compiler is incompatible, rather it leads to the version check being performed).

Reply:
Updated [65a1efc](https://github.com/PennyLaneAI/pennylane/pull/4979/commits/65a1efc6f99dd1236bc133900139803eb832e197)